url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/io/file/TestForVirtualFile.java

```diff
@@ -0,0 +1,319 @@
+			for(int i=0;i<buffer.length;i++) {
+				buffer[0] = 0;
+			}
+			int ret = vf.read(buffer);
+			assertEquals(-1, ret);
+			for(int i=0;i<buffer.length;i++) {
+				assertEquals(0, buffer[0]);
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			assertTrue(false);
+		} finally {
+			if(vf != null) {
+				vf.isCashMode(false);
+				vf.close();
+			}
+		}
+	}
+
+	
+	public void testSync() throws IOException {
+		File testPath = new File(getFile(),"__001__.txt");
+		CashKyoroFile vf = null;
+		try {
+			testPath.delete();
+			testPath.createNewFile();
+			byte[] add = {1, 2, 3, 4, 5};
+			vf = new CashKyoroFile(testPath, 1024, 5);
+			byte[] read = new byte[6];
+			vf.addChunk(add, 0, 5);
+
+			int ret = vf.read(read);
+			assertEquals(1, read[0]);
+			assertEquals(2, read[1]);
+			assertEquals(3, read[2]);
+			assertEquals(4, read[3]);
+			assertEquals(5, read[4]);
+			assertEquals(0, read[5]);
+			assertEquals(5, ret);
+		} finally {
+			if(vf != null) {
+				vf.isCashMode(false);
+				vf.close();
+			}
+		}
+	}
+
+
+	public void testUnsync() throws IOException {
+		File testPath = new File(getFile(),"__001__.txt");
+		CashKyoroFile vf = null;
+		try {
+			testPath.delete();
+			testPath.createNewFile();
+			byte[] add = {1, 2, 3, 4, 5};
+			vf = new CashKyoroFile(testPath, 1024, 5);
+			byte[] read = new byte[6];
+			vf.addChunk(add, 0, 5);
+			vf.syncWrite();
+			vf.addChunk(add, 0, 1);
+			int ret = vf.read(read);
+			assertEquals(1, read[0]);
+			assertEquals(2, read[1]);
+			assertEquals(3, read[2]);
+			assertEquals(4, read[3]);
+			assertEquals(5, read[4]);
+			assertEquals(1, read[5]);
+			assertEquals(6, ret);
+		} finally {
+			if(vf != null) {
+				vf.close();
+			}
+		}
+	}
+
+
+	public void testSecound() {
+		File testPath = new File(getFile(),"__001__.txt");
+		String testdata= 
+		 "「この間はすまなかった。いつも間が悪くて君に会う機会がない。"+
+		 "きょうは歌舞伎座の切符が二枚手に入ったから一緒に見に行か"+
+		 "ないか。午後一時の開場だから十時頃の電車で銀座あたりへ来"+
+		 "てくれるといい。君の知っているカフェーかレストランがあるだろう」";
+		CashKyoroFile vf = null;
+		try {
+			testPath.delete();
+			testPath.createNewFile();
+			vf = new CashKyoroFile(testPath, 1024, 5);
+			byte[] buffer = new byte[1000];
+			for(int i=0;i<buffer.length;i++) {
+				buffer[0] = 0;
+			}
+			int ret = vf.read(buffer);
+			assertEquals(-1, ret);
+			for(int i=0;i<buffer.length;i++) {
+				assertEquals(0, buffer[0]);
+			}
+			byte[] buff = testdata.getBytes("utf8");
+			vf.addChunk(buff, 0, buff.length);
+			{
+				// first read
+				ret = vf.read(buffer);
+				String result = new String(buffer, 0 , ret);
+				assertEquals(testdata, result);
+
+				// second read
+				{
+					for(int i=0;i<buffer.length;i++) {
+						buffer[0]=0;
+					}
+					ret = vf.read(buffer);
+					assertEquals(-1, ret);					
+				}
+			}
+			{
+				// sync
+				for(int i=0;i<buffer.length;i++) {
+					buffer[0]=0;
+				}
+				vf.syncWrite();
+				vf.seek(0);
+				ret = vf.read(buffer);
+				String result = new String(buffer, 0 , ret);
+				assertEquals(testdata, result);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 142:152, 224:234.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.